examples/basic/contact-bot.py

```diff
@@ -0,0 +1,69 @@
+"""contact bot"""
+import asyncio
+import logging
+import os
+from typing import Optional
+
+from wechaty_puppet import ScanStatus, ContactType  # type: ignore
+
+from wechaty import Wechaty, Contact
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger(__name__)
+
+
+class MyBot(Wechaty):
+    """
+    listen wechaty event with inherited functions, which is more friendly for
+    oop developer
+    """
+    def __init__(self):
+        super().__init__()
+
+    async def on_login(self, contact: Contact):
+        print(f'user: {contact} has login')
+        contacts = await self.Contact.find_all()
+        log.info('bot #######################')
+        log.info('bot Contact number: %s', len(contacts))
+
+        # official
+        for i, contact in enumerate(contacts):
+            if contact.type() == ContactType.CONTACT_TYPE_OFFICIAL:
+                log.info('Bot Official %d :%s', i, contact)
+            elif contact.type() == ContactType.CONTACT_TYPE_PERSONAL:
+                log.info('Bot Personal :%d %s %s',
+                         i, contact.name, contact.contact_id)
+
+        # get contact avatar url
+        max_num = 17
+        for i, contact in enumerate(contacts[:max_num]):
+            avatar_url = contact.payload.avatar
+            log.info(f'Bot Contact: {contact.name} with avatar url'
+                     f' {avatar_url}')
+
+        # send msg to someone
+
+        for i, contact in enumerate(contacts):
+            if contact.payload.alias == 'lover':
+                await contact.say('my love')
+
+    async def on_scan(self, status: ScanStatus, qr_code: Optional[str] = None,
+                      data: Optional[str] = None):
+        contact = self.Contact.load(self.contact_id)
+        print(f'user <{contact}> scan status: {status.name} , '
+              f'qr_code: {qr_code}')
+
+
+bot: Optional[MyBot] = None
+os.environ['WECHATY_PUPPET_HOSTIE_TOKEN'] = 'your-token-here'
```

Comment:
The goal of using an environment variable here is to prevent any hardcoding for token.

However, your code here does the hardcoding again. 

It would better to only read this environment variable from the env.

Reply:
Here is the code for setting envrioment variable, which will be automatically read. The idea of the this code is for reminding the develper, they should set enviroment variable.

Or,  all I should do, is to write doc to remind the developer they should set envrioment variable?

Reply:
`WECHATY_PUPPET_HOSTIE_TOKEN ` envrioment variable is read automatically.

Reply:
I don't think you are doing a read operation.

You are doing a write operation:

> os.environ['WECHATY_PUPPET_HOSTIE_TOKEN'] = 'your-token-here'

This means whatever the environment variable value is, you will set it to 'your-token-here', which I believe is not what we are expecting.

My suggestion would be to write how to set the environment variable in our README, and here, we just read it.

If it is not set, then show a friendly message to tell the user how to set it right.